src/StepperISR_due.cpp

```diff
@@ -1,401 +1,691 @@
+                      RISING);
+      break;
+    case 4:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(4),
+                      RISING);
+      break;
+    case 5:
+      attachInterrupt(digitalPinToInterrupt(_step_pin), ISRQueueName(5),
+                      RISING);
+      break;
+  }
+  _connected = true;
+  PWM_INTERFACE->PWM_IDR2 = 0x00FFFF0F;
+  PWM_INTERFACE->PWM_IDR1 = 0x00FFFF0F;
+}
+
+void StepperQueue::disconnect() {
+  PWMC_DisableChannel(PWM_INTERFACE, mapping->channel);
+  PWM_INTERFACE->PWM_DIS = PWM_INTERFACE->PWM_DIS & (~mapping->channelMask);
+  _connected = false;
+}
+
+void StepperQueue::commandAddedToQueue(bool start) {
+  struct queue_entry* e = &entry[read_idx & QUEUE_LEN_MASK];
+  interrupts();
+  if (e->toggle_dir) {
+    e->toggle_dir = 0;
+    *_dirPinPort ^= _dirPinMask;
+    delayMicroseconds(30);
+  }
+  if (start) {
+    noInterrupts();
+    bool first = (next_write_idx++ == read_idx);
+    interrupts();
+    if (!first) {
+      if (e->steps > 0 || e->hasSteps) {
+        if (!_hasISRactive) {
+          _hasISRactive = true;
+          _runOnce = false;
+          _skipNextPWMInterrupt = false;
+          PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CPRD = e->ticks;
+          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
+                              true);
+        }
+        return;
+      } else {
+        if (!_hasISRactive) {
+          _hasISRactive = true;
+          PIO_SetOutput(mapping->port, g_APinDescription[mapping->pin].ulPin, 0,
+                        0, 0);
+          if (PWM_INTERFACE->PWM_SR & (1 << mapping->channel)) {
+            PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CPRD = e->ticks;
+          } else {
+            PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CPRDUPD = e->ticks;
+          }
+          /*Disconnect, but do not disable, enable the PWM ISR*/
+          mapping->port->PIO_IDR = g_APinDescription[mapping->pin].ulPin;
+          PIO_SetOutput(mapping->port, g_APinDescription[mapping->pin].ulPin, 0,
+                        0, 0);
+          PWM_INTERFACE->PWM_ENA |= mapping->channelMask;
+          _pauseCommanded = true;
+          /*Enable the ISR too so we don't miss it*/
+          PWM_INTERFACE->PWM_IER1 =
+              PWM_INTERFACE->PWM_IMR1 | mapping->channelMask;
+          return;
+        } else {
+          // The ISR should handle this case...
+        }
+      }
+      _connected = true;
+    }
+  }
+}
+
+int8_t StepperQueue::startPreparedQueue() { return AQE_OK; }
+
+void StepperQueue::forceStop() {
+  noInterrupts();
+  read_idx = next_write_idx;
+  interrupts();
+  PWMC_DisableChannel(PWM_INTERFACE, mapping->channel);
+}
+
+bool StepperQueue::isValidStepPin(uint8_t step_pin) {
+  if (pinToChannel(step_pin) < 0x08) {
+    if (!channelsUsed[pinToChannel(step_pin)]) {
+      channelsUsed[pinToChannel(step_pin)] = true;
+      numChannels++;
+      return true;
```

Comment:
The channel allocation should not be here. This function is only a test.

Reply:
hmmmm....I don't even remember doing that, but yes, I think that was supposed to be in the init function...With this, I decided to make all the changes and will test as soon as I get to the bottom of the list.  Obviously I spent more time on the ISRs than the support code.  :)